packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -437,31 +445,28 @@ export function createRootLayoutValidatorStream(): TransformStream<
   let foundHtml = false
   let foundBody = false
 
-  const decoder = new TextDecoder()
-
-  let content = ''
+  let content = new Uint8Array(0)
   return new TransformStream({
     async transform(chunk, controller) {
+      content = concatUint8Arrays(content, chunk)
       // Peek into the streamed chunk to see if the tags are present.
       if (!foundHtml || !foundBody) {
-        content += decoder.decode(chunk, { stream: true })
-        if (!foundHtml && content.includes('<html')) {
+        if (
+          !foundHtml &&
+          indexOfUint8Array(content, ENCODED_TAGS.OPENING.HTML)
+        ) {
           foundHtml = true
         }
-        if (!foundBody && content.includes('<body')) {
+        if (
+          !foundBody &&
+          indexOfUint8Array(content, ENCODED_TAGS.OPENING.BODY)
+        ) {
           foundBody = true
         }
       }
       controller.enqueue(chunk)
     },
     flush(controller) {
-      // Flush the decoder.
-      if (!foundHtml || !foundBody) {
-        content += decoder.decode()
-        if (!foundHtml && content.includes('<html')) foundHtml = true
-        if (!foundBody && content.includes('<body')) foundBody = true
-      }
```

Comment:
I don't think we have to do this anymore since we are no longer using a decoder.